client/src/utils/jobLeads.js

```diff
@@ -0,0 +1,87 @@
+// helper functions with all things related to job Leads
+
+function valuetext(value) {
+  return `$${value}`;
+}
+
+function findCompensationRange(managedJobLeads) {
+  if (managedJobLeads.length < 1) return [null, null];
+
+  let minCompensation = managedJobLeads[0]?.compensationMin ?? 0;
+  let maxCompensation = managedJobLeads[0]?.compensationMax ?? 0;
+
+  managedJobLeads.forEach((lead) => {
+    if (lead.compensationMin < minCompensation)
+      minCompensation = lead.compensationMin;
+    if (lead.compensationMax > maxCompensation)
+      maxCompensation = lead.compensationMax;
+  });
+
+  if (minCompensation === maxCompensation) {
```

Comment:
Can we add a check if any compensation is null we set it to 0. It tends to break the table page. This might have to be on the backend